lib/commands.coffee

```diff
@@ -60,21 +61,25 @@ module.exports =
 				stdio: [ process.stdin, 'pipe', 'pipe' ]
 
 	burn: (image, operation, options) ->
-
 		# Default image to the given path
-		operation.image ?= image
+		image = operation.image ? image
 
 		Promise.try ->
 			if not options?.drive?
 				throw new Error('Missing drive option')
 
-			return operation.image
-		.then(fs.statAsync).get('size')
-		.then (size) ->
-			imageReadStream = fs.createReadStream(operation.image)
-
-			# This is read by Resin Image Write to
-			# emit correct `progress` events.
-			imageReadStream.length ?= size
-
-			return imageWrite.write(options.drive, imageReadStream)
+			Promise.props
+				imageSize: fs.statAsync(image).get('size')
+				imageStream: Promise.resolve(fs.createReadStream(image))
```

Comment:
Nit: You don't need a `Promise.resolve` here, Bluebird will happily treat bare values as if they'd come from promises.